Content.Server/StationEvents/Events/GreytideVirusRule.cs

```diff
@@ -57,12 +61,16 @@ protected override void Started(EntityUid uid, GreytideVirusRuleComponent virusC
         var firelockQuery = GetEntityQuery<FirelockComponent>();
         var accessQuery = GetEntityQuery<AccessReaderComponent>();
 
-        var lockQuery = AllEntityQuery<LockComponent>();
-        while (lockQuery.MoveNext(out var lockUid, out var lockComp))
+        var lockQuery = AllEntityQuery<LockComponent, TransformComponent>();
+        while (lockQuery.MoveNext(out var lockUid, out var lockComp, out var xform))
         {
             if (!accessQuery.TryComp(lockUid, out var accessComp))
                 continue;
 
+            // make sure not to hit CentCom or other maps
+            if (CompOrNull<StationMemberComponent>(xform.GridUid)?.Station != chosenStation)
+                continue;
+
```

Comment:
I'm pretty sure the event runner is tied to the station so we should really update the method to pass in the station itself if possible?